src/coreclr/jit/lower.cpp

```diff
@@ -3165,8 +3166,58 @@ void Lowering::LowerStoreLocCommon(GenTreeLclVarCommon* lclStore)
                 return;
             }
 #endif // !WINDOWS_AMD64_ABI
+            convertToStoreObj = false;
         }
-        else if (!src->OperIs(GT_LCL_VAR) || (varDsc->GetLayout()->GetRegisterType() == TYP_UNDEF))
+        else if (!varDsc->IsEnregisterable())
+        {
+            convertToStoreObj = true;
+        }
+        else if (src->OperIs(GT_CNS_INT))
+        {
+            assert(src->IsIntegralConst(0) && "expected an INIT_VAL for non-zero init.");
+            var_types regType = varDsc->GetRegisterType();
+#ifdef FEATURE_SIMD
+            if (varTypeIsSIMD(regType))
+            {
+                if (!varDsc->lvDoNotEnregister)
+                {
+                    CorInfoType simdBaseJitType = comp->getBaseJitTypeOfSIMDLocal(lclStore);
+                    if (simdBaseJitType == CORINFO_TYPE_UNDEF)
+                    {
+                        // Lie about the type if we don't know/have it.
+                        simdBaseJitType = CORINFO_TYPE_FLOAT;
+                    }
+                    GenTreeSIMD* simdTree =
+                        comp->gtNewSIMDNode(regType, src, SIMDIntrinsicInit, simdBaseJitType, varDsc->lvExactSize);
+                    BlockRange().InsertAfter(src, simdTree);
```

Comment:
Shouldn't we `lower` the newly created `simdTree` so containment and other transforms can happen as appropriate?

Reply:
Nice catch, thank you! It fixes a todo that I added because now we have SIMD16 instead of SIMD12.